src/coverlet.collector/DataCollector/AttachmentManager.cs

```diff
@@ -0,0 +1,156 @@
+        public void SendCoverageReport(string coverageReport)
+        {
+            // Save coverage report to file
+            var coverageReportPath = this.SaveCoverageReport(coverageReport);
+
+            // Send coverage attachment to test platform.
+            this.SendAttachment(coverageReportPath);
+        }
+
+        /// <summary>
+        /// Disposes attachment manager
+        /// </summary>
+        public void Dispose()
+        {
+            // Unregister events
+            if (this.dataSink != null)
+            {
+                this.dataSink.SendFileCompleted -= this.OnSendFileCompleted;
+            }
+        }
+
+        /// <summary>
+        /// Saves coverage report to file system
+        /// </summary>
+        /// <param name="report">Coverage report</param>
+        /// <returns>Coverage report file path</returns>
+        private string SaveCoverageReport(string report)
+        {
+            try
+            {
+                this.directoryHelper.CreateDirectory(this.reportDirectory);
+                var filePath = Path.Combine(this.reportDirectory, this.reportFileName);
+                this.fileHelper.WriteAllText(filePath, report);
+                this.eqtTrace.Info("{0}: Saved coverage report to path: '{1}'", CoverletConstants.DataCollectorName, filePath);
+
+                return filePath;
+            }
+            catch (Exception ex)
+            {
+                var errorMessage = string.Format(Resources.FailedToSaveCoverageReport, CoverletConstants.DataCollectorName, this.reportFileName, this.reportDirectory);
+                throw new CoverletDataCollectorException(errorMessage, ex);
+            }
+        }
+
+        /// <summary>
+        /// SendFileCompleted event handler
+        /// </summary>
+        /// <param name="sender">Sender</param>
+        /// <param name="e">Event args</param>
+        public void OnSendFileCompleted(object sender, AsyncCompletedEventArgs e)
+        {
+            try
+            {
+                this.eqtTrace.Verbose("{0}: SendFileCompleted received", CoverletConstants.DataCollectorName);
+                this.CleanupReportDirectory();
+            }
+            catch (Exception ex)
+            {
+                this.logger.LogWarning(ex.ToString());
+                this.Dispose();
+            }
+        }
+
+        /// <summary>
+        /// Sends attachment file to test platform
+        /// </summary>
+        /// <param name="attachmentPath">Attachment file path</param>
+        private void SendAttachment(string attachmentPath)
+        {
+            if (this.fileHelper.Exists(attachmentPath))
+            {
+                // Send coverage attachment to test platform.
+                this.eqtTrace.Verbose("{0}: Sending attachment to test platform", CoverletConstants.DataCollectorName);
+                this.dataSink.SendFileAsync(this.dataCollectionContext, attachmentPath, false);
+            }
```

Comment:
`attachmentPath` should exits at this point...but add some verbose logs on else?

Reply:
Added log for this